File: app/src/main/java/fr/smarquis/soundquicksettings/SoundTileService.java
package fr.smarquis.soundquicksettings;

import android.annotation.TargetApi;
import android.content.Context;
import android.content.Intent;
import android.media.AudioManager;
import android.os.Build;
import android.service.quicksettings.TileService;

import static android.media.AudioManager.ADJUST_SAME;
import static android.media.AudioManager.FLAG_SHOW_UI;

@TargetApi(Build.VERSION_CODES.N)
public class SoundTileService extends TileService {

    @Override
    public void onClick() {
        super.onClick();
        ((AudioManager) getSystemService(Context.AUDIO_SERVICE)).adjustVolume(ADJUST_SAME, FLAG_SHOW_UI);
        sendBroadcast(new Intent(Intent.ACTION_CLOSE_SYSTEM_DIALOGS));
    }
}

File: app/src/main/java/fr/smarquis/soundquicksettings/SoundTileService.kt
package fr.smarquis.soundquicksettings

import android.app.AlertDialog
import android.content.DialogInterface
import android.media.AudioManager
import android.media.AudioManager.ADJUST_SAME
import android.media.AudioManager.FLAG_SHOW_UI
import android.service.quicksettings.TileService
import android.view.WindowManager.LayoutParams.FLAG_DIM_BEHIND

class SoundTileService : TileService() {
    override fun onClick() {
        getSystemService(AudioManager::class.java).adjustVolume(ADJUST_SAME, FLAG_SHOW_UI)
        AlertDialog.Builder(this).create().apply {
            window?.clearFlags(FLAG_DIM_BEHIND)
            setOnShowListener(DialogInterface::dismiss)
        }.let(::showDialog)
    }
}
